### PR TITLE
Add node/version to the user agent header.

### PR DESCRIPTION
**Pull request by emilyashley**
## Which problem is this PR solving?

- closes https://github.com/honeycombio/libhoney-js/issues/298

## Short description of the changes

- Add node version to user-agent header. 
- Test expected versions are in the header and that it is in the correct order

Screen Shot 2022-09-13 at 12 25 05 PM

### CODE REVIEW

**Comment by robbkidd** [comment]
> Does the team have opinions on if it's `node/v#.#.#` or `node/#.#.#`

I _wish_ for no-v (`node/#.#.#`), but we have other user-agents incoming with "v" in the version number which means we have to accommodate them anyway, so I'd say whichever you already have working.

**Review by pkanal — APPROVED** [review]
Nice! I think we should keep the string to whatever process.version is

File: src/__tests__/transmission_test.js [hunk 1]
```diff
@@ -530,15 +530,18 @@ describe("base transmission", () => {
...
+          userAgent.indexOf(`node/${process.version}`) > 1 &&
           userAgent.indexOf("addition") === -1
       },
       {
         dataset: "test-transmission2",
         addition: "user-agent addition",
         probe: userAgent =>
-          userAgent.indexOf("libhoney") === 0 &&
-          userAgent.indexOf("addition") !== -1
+        // user-agent order: libhoney, addition, node
```
Comment (robbkidd) [inline src/__tests__/transmission_test.js:542]: ✨ intention revealing commentary ✨ 

Terrific

**Review by robbkidd — APPROVED** [review]
Join the JS feista!

**Pull request merged**